tekton/integration_pipeline.go

```diff
@@ -125,15 +152,97 @@ func NewIntegrationPipelineRun(prefix, namespace string, integrationTestScenario
 		Spec: tektonv1.PipelineRunSpec{
 			PipelineRef: &tektonv1.PipelineRef{
 				ResolverRef: tektonv1.ResolverRef{
-					Resolver: tektonv1.ResolverName(integrationTestScenario.Spec.ResolverRef.Resolver),
-					Params:   resolverParams,
+					Resolver: tektonv1.ResolverName(resolver),
+					Params:   tektonResolverParams,
 				},
 			},
 		},
 	}
+
 	return &IntegrationPipelineRun{pipelineRun}
 }
 
+func getPipelineRunYamlFromPipelineRunResolver(client client.Client, ctx context.Context, prefix, namespace, resolver string, resolverParams []v1beta2.ResolverParameter) (string, error) {
+	stringResolverParams := map[string]string{}
+
+	for _, scenarioParam := range resolverParams {
+		stringResolverParams[scenarioParam.Name] = scenarioParam.Value
+	}
+
+	request := resolutionv1alpha1.ResolutionRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: prefix + "-",
+			Namespace:    namespace,
+			Labels: map[string]string{
+				"resolution.tekton.dev/type": resolver,
+			},
+			Annotations: map[string]string{
+				"konflux-ci.dev/created-by": "integration-service", // for backup garbage collection
+			},
+		},
+		Spec: resolutionv1alpha1.ResolutionRequestSpec{
+			Parameters: stringResolverParams,
+		},
+	}
+
+	err := retry.OnError(retry.DefaultBackoff, func(e error) bool { return true }, func() error {
+		// An error is occurring in here
+		// no kind is registered for the type v1alpha1.ResolutionRequest in scheme "k8s.io/client-go/kubernetes/scheme/register.go:80"
+		return client.Create(ctx, &request)
+	})
+	if err != nil {
+		return "", err
+	}
+	resolutionRequestName := request.ObjectMeta.Name
+	defer func() {
+		_ = retry.OnError(retry.DefaultBackoff, func(e error) bool { return true }, func() error {
+			return client.Delete(ctx, &request)
+		})
+	}()
+
+	resolverBackoff := wait.Backoff{
+		Steps:    60,
+		Duration: 1 * time.Second,
+		Factor:   1.0,
+		Jitter:   0.5,
+	}
+	var resolvedRequest resolutionv1alpha1.ResolutionRequest
+	err = retry.OnError(resolverBackoff, func(e error) bool { return true }, func() error {
+		err = client.Get(ctx, types.NamespacedName{Namespace: namespace, Name: resolutionRequestName}, &resolvedRequest)
+		if err != nil {
+			return err
+		}
+		for _, cond := range resolvedRequest.Status.Conditions {
+			if cond.Type == knative.ConditionSucceeded {
+				if cond.Status == corev1.ConditionTrue {
+					// request resolved successfully, we can terminate retry block
+					return nil
+				}
+			}
+		}
+		return fmt.Errorf("resolution for '%s' in namespace '%s' did not complete", namespace, resolutionRequestName)
+	})
+
+	if err != nil {
+		return "", err
+	}
+	return resolvedRequest.Status.Data, nil
+}
+
+func generateIntegrationPipelineRunFromBase64(base64plr string) (*IntegrationPipelineRun, error) {
+	plrYaml, err := base64.StdEncoding.DecodeString(base64plr)
+	if err != nil {
+		return nil, err
+	}
+	// TODO: Make sure this is sufficient to verify that kind is pipelineRun
```

Comment:
It might be a good idea to create a follow-up ticket for this.
I'm a bit worried that the current approach may cause issues if there are major differences between the tekton version that we use in the integration service code vs the one that's deployed on the cluster.
Since we're supposed to keep pace with it anyways, it's probably not a major concern, but it may be good to have a tracker for it to improve in the future.

Reply:
That was actually a note for myself.  I verified and forgot to remove the comment. Unless there's some other CRD  with identical fields to the pipelinerun and a different controller this will be fine.  I think that's unlikely enough that we can just use unmarshalling as verification